internal/database/txn/transaction.go

```diff
@@ -167,39 +182,68 @@ func (t *RetryingTxnRunner) StdTxn(ctx context.Context, db *sql.DB, fn func(cont
 	return t.run(ctx, func(ctx context.Context) error {
 		tx, err := db.BeginTx(ctx, nil)
 		if err != nil {
+			// This was lifted from the LXD code.
+			// It has been observed that we can get into a strange state when
+			// Dqlite is busy performing a checkpoint operation.
+			// This is an attempt to save an otherwise poisoned database.
+			if strings.Contains(err.Error(), txnInTxn) {
+				_, _ = db.Exec("ROLLBACK")
```

Comment:
Shouldn't we at least log the error when trying to rollback?

Reply:
If you can't roll back here, you're hosed for a while. You'll get a burst of transaction-in-transaction errors. It will be obvious.